src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2601,6 +2627,8 @@ private void onDecodeBegin(
                 long authorization)
             {
                 doEncodeBegin(traceId, authorization, incomingChannel, nextOutgoingId);
+                sessionState = sessionState.sentBegin();
+                assert sessionState != AmqpSessionState.ERROR;
```

Comment:
Could these be moved to a session-scoped `doEncodeBegin(traceId, authorization)` with fewer arguments instead?
Then it doesn't matter why `BEGIN` was sent, the state machine always represents the session state accurately.

Reply:
Do you mean to create a method `doEncodeBegin(traceId, authorization)` in `AmqpSession`, and then move those 3 lines to that method
```
doEncodeBegin(traceId, authorization, incomingChannel, nextOutgoingId);
sessionState = sessionState.sentBegin();
assert sessionState != AmqpSessionState.ERROR;
```
and then call the `doEncodeBegin(traceId, authorization)` method inside this `onDecodeBegin()` method?

Reply:
Sounds good, yes!